src/compiler/core.ts

```diff
@@ -1187,7 +1198,8 @@ namespace ts {
     /**
      *  List of supported extensions in order of file resolution precedence.
      */
-    export const supportedTypeScriptExtensions = [".ts", ".tsx", ".d.ts"];
+    export const supportedTypeScriptExtensionsNonDts = [".ts", ".tsx"];

```

Comment:
I don't think you'll need this if you correctly support errors for importing `.d.ts` files